pdata/pmetric/alias.go

```diff
@@ -54,25 +54,9 @@ const (
 // It describes how those values relate to the time interval over which they are aggregated.
 type MetricDataPointFlags = internal.MetricDataPointFlags
 
-const (
-	// MetricDataPointFlagsNone is the default MetricDataPointFlags.
-	MetricDataPointFlagsNone = internal.MetricDataPointFlagsNone
-)
-
-// NewMetricDataPointFlags returns a new MetricDataPointFlags combining the flags passed
-// in as parameters.
+// NewMetricDataPointFlags returns a new empty NewMetricDataPointFlags.
 var NewMetricDataPointFlags = internal.NewMetricDataPointFlags
 
-// MetricDataPointFlag allow users to configure DataPointFlags. This is achieved via NewMetricDataPointFlags.
-// The separation between MetricDataPointFlags and MetricDataPointFlag exists to prevent users accidentally
-// comparing the value of individual flags with MetricDataPointFlags. Instead, users must use the HasFlag method.
-type MetricDataPointFlag = internal.MetricDataPointFlag
-
-const (
-	// MetricDataPointFlagNoRecordedValue is flag for a metric aggregator which reports changes since last report time.
-	MetricDataPointFlagNoRecordedValue = internal.MetricDataPointFlagNoRecordedValue
-)
-
```

Comment:
We need to go via deprecation, cannot directly delete the code.

Reply:
Agreed.  I will get started on that now that the end state is finalized.

Reply:
Re-added the original solution and added a new field to pdata called `FlagsStruct` that exposes the new capabilities.